### PR TITLE
Add color to Category

### PR DESCRIPTION
**Pull request by leungkenneth**
image

@HeinzUBC this is the only change that I'm suggesting: it would involve no change on your side. This would eliminate a lot of code smell associated with ChartSettings controller which currently needs code to sync up the list of colors with which category they belong to and which user they belong to.

@AlanZhangDev could also use this as well and add a reference to a category to habits (so habits have color on the Gantt Chart). Whether or not he wants to do this before or after the 29th is up to him.

### CODE REVIEW

File: server/models/Category.model.js [hunk 1]
```diff
@@ -11,6 +11,10 @@ const categorySchema = new mongoose.Schema({
     type: String,
     required: true,
     trim: true,
+  },
+  color: {
+    type: String,
+    default: '#000000'
   }
```
Comment (HeinzUBC) [inline server/models/Category.model.js:18]: I think we can first have my bigger pull-request merged in before this

Comment (HeinzUBC) [reply server/models/Category.model.js:18]: In terms of my initializer data for todo items, currently none of them have this field. Do I need to manually add this field to each of my initializer json todo objects, or will they acquire this field as soon as I import the json file into the todo item collection in MongoDB?

**Comment by leungkenneth** [comment]
PR converted to draft since it seems this change will not be used by tomorrow.

**Review by HeinzUBC — APPROVED** [review]

**Review by HeinzUBC — CHANGES_REQUESTED** [review]
paused

File: server/models/Category.model.js (hunk 1 above)
Comment (leungkenneth) [reply server/models/Category.model.js:18]: If the declared document doesn't have this field it should be added with its default value so there should be no changes required from your side.

**Review by HeinzUBC — APPROVED** [review]

**Pull request merged**